### PR TITLE
Expose redaction power level in room settings

### PR DESCRIPTION
**Pull request by SimonBrandner**
Type: enhancement
Fixes https://github.com/vector-im/element-web/issues/20590

---
Here's what your changelog entry will look like:

## ✨ Features
 * Expose redaction power level in room settings ([\#7599](https://github.com/matrix-org/matrix-react-sdk/pull/7599)). Fixes vector-im/element-web#20590. Contributed by @SimonBrandner.

Preview: https://pr7599--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Review by turt2live — APPROVED** [review]
code looks fine, but I think this requires product and/or design to look at it - there's potential confusion between this setting and `redacts`, which achieve very different things.

There's also potential for concerns around making it easy for people to block self-redactions in rooms from a product perspective. We might be better off inventing a UI flow for making announcement-only rooms instead.

File: src/components/views/settings/tabs/room/RolesRoomSettingsTab.tsx [hunk 1]
```diff
@@ -237,6 +238,7 @@ export default class RolesRoomSettingsTab extends React.Component<IProps> {
             [EventType.RoomEncryption]: _td("Enable room encryption"),
             [EventType.RoomServerAcl]: _td("Change server ACLs"),
             [EventType.Reaction]: _td("Send reactions"),
+            [EventType.RoomRedaction]: _td("Redact messages"),
```
Comment (t3chguy) [inline src/components/views/settings/tabs/room/RolesRoomSettingsTab.tsx:241]: the string should be `Remove`

image

**Review by t3chguy — CHANGES_REQUESTED** [review]

**Comment by kittykat** [comment]
@SimonBrandner can you share a screenshot of what this will look like in in the UI please?

**Review by kittykat — CHANGES_REQUESTED** [review]
Can you please move the new setting to be above "Remove messages sent by others" and change the string from "Remove messages" to "Remove your own messages"?

Otherwise LGTM

**Comment by SimonBrandner** [comment]
From a quick look, adding some sort of ordering might be a bigger change, so I'd rather leave it for a separate PR

**Comment by SimonBrandner** [comment]
Updated image

**Review by t3chguy — APPROVED** [review]

**Review by kittykat — APPROVED** [review]

**Pull request merged**